### PR TITLE
Add .travis.yml

### LINKED ISSUE #1913: Setup Travis
Because CI is nice

_Originally posted by @bhousel in https://github.com/osmlab/name-suggestion-index/pull/1910/comment#issuecomment-430729658_

### PR DESCRIPTION
**Pull request by ENT8R**
Fixes #1913

I assume that you don't want to receive any emails, so I just disabled notifications... You just need to go to https://travis-ci.org/osmlab/name-suggestion-index and enable Travis for this repo. But I think you know what you have to do 😄

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,14 +19,15 @@
...
+    "wikimatch": "node match_wikiTags",
+    "test": "node validate"
   },
   "devDependencies": {
     "clear": "^0.1.0",
     "colors": "^1.1.2",
     "diacritics": "^1.3.0",
     "json-stringify-pretty-compact": "^1.1.0",
-    "jsonschema": "^1.1.0",
+    "jsonschema": "^1.2.4",
```
Comment (good-praxis) [inline package.json]: Probably/Shouldn't break anything, but why the update?

File: validate.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            if (e.property) {
+                console.error('  ' + colors.yellow(e.property + ' ' + e.message));
+            } else {
+                console.error('  ' + colors.yellow(e));
+            }
+        });
+        console.error();
+        process.exit(1);
+    }
+}
```
Comment (good-praxis) [inline validate.js]: Good idea, you should add a build test, and validate that the newly-build files are the same that were committed!

File: package.json (hunk 1 above)
Comment (ENT8R) [reply package.json]: Why not? It is always good to use the latest version to avoid security issues...

File: validate.js (hunk 2 above)
Comment (ENT8R) [reply validate.js]: I don't really get what you mean... When should this script run?

File: package.json (hunk 1 above)
Comment (good-praxis) [reply package.json]: Fair point!

File: validate.js (hunk 2 above)
Comment (good-praxis) [reply validate.js]: Well, the build only functions if the json is valid anyway (i.e. see what `npm run build` does) - and if the contributor actually ran the build locally and committed the results, the build files (`/dist/`) shouldn't change after the build.

This travis config will already prevent an array of problems, though, so good job!

**Comment by matkoniecz** [comment]
@bhousel Can you look at this? I see no problems, but in this PR I would not notice if something would be wrong.

**Comment by bhousel** [comment]
Oh yeah - it seems fine!  Looks like it just does the schema validation step only, which is good 👍

**Pull request merged**